### DISCUSSION TITLE
Error compiling on Arm64 ubuntu 20.04 - filcrypto.pc': No such file or directory

### DISCUSSION THREAD

**Discussion by Digital-Nomad** [filecoin-project/lotus#6619] [General Q&A]
> Note: For security-related bugs/issues, please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).

**Describe the bug**
I am getting the following error when I try to compile lotus with ARM64 Ubuntu 20.04

```
go: downloading github.com/filecoin-project/go-padreader v0.0.0-20200903213702-ed5fae088b20
go: downloading github.com/filecoin-project/go-statemachine v0.0.0-20200925024713-05bd7c71fbfe
# pkg-config --cflags  -- /home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc /home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc /home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc /home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc /home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc
Failed to open '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc': No such file or directory
No package '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc' found
Failed to open '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc': No such file or directory
No package '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc' found
Failed to open '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc': No such file or directory
No package '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc' found
Failed to open '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc': No such file or directory
No package '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc' found
Failed to open '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc': No such file or directory
No package '/home/simon/GolandProjects/lotus/extern/filecoin-ffi/generated/../filcrypto.pc' found
pkg-config: exit status 1
```
Any ideas?

**Comment by xxccli** [comment]
': No such file or directory

**Comment by jennijuju** [comment]
`git submodule update --init --recursive`?

**Reply by xloem** [reply]
Hey I'm having this issue, and I can work around it when building from the git tree, but I'm not sure how to handle it when something else depends on lotus with `go get`.  Edit: I've fixed this, see follow-up post

**Comment by xloem** [comment]
It looks like filecoin-ffi needs to be handled specially in every project that depends on it, no matter how distantly, and that process can kind of get messed up sometimes.

This worked for me:

Build filecoin-ffi from source inside its extern/filecoin-ffi subfolder: `make clean ; FFI_BUILD_FROM_SOURCE=1 make`
https://github.com/filecoin-project/filecoin-ffi#building-from-source

If developing your own code, replace filecoin-ffi inside the local go.mod to point to a submodule inside e.g. extern/filecoin-ffi: ` go mod edit -replace=github.com/filecoin-project/filecoin-ffi=./extern/filecoin-ffi` https://github.com/filecoin-project/filecoin-ffi#go-get

**Comment by LexLuthr** [comment]
@Digital-Nomad Do you still need help with this or was this resolved? If it was resolved, can you please tell how you fixed it?
